File: Tugas Modul 4 (ROS)/bev_workspace/src/bev_package/src/service_server_node.cpp
#include "ros/ros.h"
#include "bev_package/service.h"

bool magnitude_formula(bev_package::service::Request &req, bev_package::service::Response &res){
  double magnitude = sqrt(pow(req.komponenX, 2) + pow(req.komponenY, 2) + pow(req.komponenZ, 2));
  ROS_INFO("Nilai magnitude: %f", magnitude);
  return true;
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "service_server");
  ros::NodeHandle nh;
  ros::ServiceServer service = nh.advertiseService("service", magnitude_formula);

  ROS_INFO("Service siap dijalankan.");

  ros::spin();

  return 0;
}

File: Tugas Modul 4 (ROS)/bev_workspace/src/bev_package/script/service_server_node.py
import rospy
from bev_package.srv import service, serviceResponse

def magnitude_formula(req):
    magnitude = (req.komponenX ** 2 + req.komponenY ** 2 + req.komponenZ ** 2) ** 0.5
    rospy.loginfo("Nilai magnitude: %f", magnitude)
    return serviceResponse(magnitude)

if __name__ == '__main__':
    rospy.init_node('service_server')
    service = rospy.Service('service', service, magnitude_formula)
    rospy.loginfo("Service siap dijalankan.")
    rospy.spin()

